tests/test_helper/bats/ovsdb_schema_upgrade.bats

```diff
@@ -63,24 +63,31 @@ setup() {
     local containers_size="${#containers_to_upgrade[@]}"
     local upgraded_size=0
 
+    # Loop below detects and performs internal dqlite schema upgrade if necessary.
+    # When upgrading dqlite, all nodes have to be upgraded before cluster becomes
+    # usable again.
     for ((container_index = 0; container_index < "$containers_size"; container_index++)) do
         local container="${containers_to_upgrade[$container_index]}"
         # Upgrade MicroOVN cluster, one cluster member at a time.
         install_microovn "$MICROOVN_SNAP_PATH" "$container"
         upgraded_size=$(( "$upgraded_size" + 1 ))
 
-        # Check if the node is fully online. If it's not, then we have a dqlite schema upgrade and need to upgrade all nodes first.
-        # If all nodes are reporting ONLINE, then we can move on to check the ovsdb schema before upgrading all nodes.
-        wait_ovn_services "$container"
-        run wait_microovn_online "$container" 30
-        if [ "$status" = 0 ] ; then
-            echo "# ($container) reports all systems are reachable. No dqlite schema update detected"
-            break
+        # After the first host is upgraded, we can try to determine whether internal Dqlite schema upgrade
+        # is required.
+        if [ "$container_index" -eq 0 ]; then
+            wait_ovn_services "$container"
+            run wait_microovn_online "$container" 30
+            # If all nodes show up as ONLINE we can assume that Dqlite upgrade is not necessary
+            if [ "$status" = 0 ] ; then
+                echo "# ($container) reports all systems are reachable. No dqlite schema update detected" >&3
+                break
+            fi
         fi
 
         # If we upgraded all systems because of a dqlite schema update, ensure all nodes are online.
         if [ "$upgraded_size" -eq "$containers_size" ] ; then
-            grep -q 0 <<< "$status"
+            run wait_microovn_online "$container" 45
```

Comment:
we are sure that 45s timeout is proving to be enough time, correct?

Reply:
I gotta admit, I'm still not fully clear on why was this occasionally failing. When trying manually, I see cluster transition form `UPGRADING/NEEDS UPGRADE` to `ONLINE` almost instantly, every time. Nevertheless, i believe this new approach is sufficient for us as it explicitly tests that after upgrading every node, they become online within 45 seconds, and it looks to be stable in CI.